test/FunctionalTests/Microsoft.OData.Core.Tests/ODataNotificationReaderTests.cs

```diff
@@ -0,0 +1,161 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataNotificationReaderTests.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.OData.Tests
+{
+    public class ODataNotificationReaderTests
+    {
+        private MemoryStream stream;
+        private TextReader reader;
+        private IODataStreamListener streamListener;
+
+        public ODataNotificationReaderTests()
+        {
+            this.stream = new MemoryStream();
+            this.reader = new StreamReader(this.stream);
+            this.streamListener = new MockODataStreamListener(new StreamWriter(this.stream));
+        }
+
+        [Theory]
+        [InlineData(true, "StreamDisposed")]
+        [InlineData(false, "StreamDisposedAsync")]
+        public void NotificationReaderDisposeShouldInvokeStreamDisposed(bool synchronous, string expected)
+        {
+            // We care about the notification reader being disposed
+            // We don't care about the reader passed to the notification reader
+            using (var notificationReader = new ODataNotificationReader(
+                this.reader,
+                this.streamListener,
+                synchronous))
+            {
+            }
+
+            var result = ReadStreamContents();
+
+            Assert.Equal(expected, result);
+        }
+
+        [Theory]
+        [InlineData(true, "StreamDisposed")]
+        [InlineData(false, "StreamDisposedAsync")]
+        public void NotificationReaderDisposeShouldBeIdempotent(bool synchronous, string expected)
+        {
+            var notificationReader = new ODataNotificationReader(
+                this.reader,
+                this.streamListener,
+                synchronous);
+
+            // 1st call to Dispose
+            notificationReader.Dispose();
+            // 2nd call to Dispose
+            notificationReader.Dispose();
+
+            var result = ReadStreamContents();
+
+            // StreamDisposed/StreamDisposeAsync was written only once
+            Assert.Equal(expected, result);
+        }
+
+#if NETCOREAPP3_1

```

Comment:
Why not `NETSTANDARD2_0` since that's what you used on the class definition?

Reply:
Because `NETSTANDARD2_0` is not a target framework in the test project (Microsoft.OData.Core.Tests) but it is in Microsoft.OData.Core project.
`await using` is only supported in `NETCOREAPP3_1` or higher